expyfun/stimuli/_hrtf.py

```diff
@@ -49,20 +62,56 @@ def _get_hrtf(angle, source, fs):
     if angle < 0:
         leftward = True
         read_angle = -angle
-    if read_angle not in angles:
+    if read_angle not in angles and not interp:
         raise ValueError('angle "{0}" must be one of +/-{1}'
                          ''.format(angle, list(angles)))
     brir = data['brir']
-    idx = np.where(angles == read_angle)[0]
-    assert len(idx) == 1
-    brir = brir[idx[0]].copy()
+    if not interp:
+        idx = np.where(angles == read_angle)[0]
+        assert len(idx) == 1
+        brir = brir[idx[0]].copy()
+    elif interp:
+        idx = np.where(angles < read_angle)[0]
+        # angles are
+        b = angles[idx[-1]]
+        c = angles[idx[-1] + 1]
+        # get known brirs
+        brir_b =  brir[idx[-1]].copy()
+        brir_c = brir[idx[-1] + 1].copy()
+    
+        # find location of maximum for each component of each known brir and
+        # average to find a generally good place to declare time = 0 (all data
+        # from before this point gets shifted to the end)
+        delay = int(np.mean([np.argmax(brir_b, 1), np.argmax(brir_c, 1)]))
+        brir_b = np.concatenate((brir_b[:, delay:], brir_b[:, :delay]), 1)
+        brir_c = np.concatenate((brir_c[:, delay:], brir_c[:, :delay]), 1)
+        # convert to frequency domain representation
+        HRTF_b = np.fft.fft(brir_b)
+        HRTF_c = np.fft.fft(brir_c)
+
+        # weighted averages of log magnitude and unwrapped phase
+        step = c-b
+        a = float(angle)
+        weight_b = (step - np.abs(a - b)) / step
+        weight_c = (step - np.abs(a - c)) / step
+        HRTF_a_logmag = (weight_b * np.log10(np.abs(HRTF_b)) + 
```

Comment:
I think it should be computationally faster do do the geometric mean rather than the mean of the logs.

So: ``HRTF_a_mag = np.abs(HRTF_b) ** weight_b * np.abs(HRTF_c) ** weight_c``